pkg/kubectl/cmd/testing/fake.go

```diff
@@ -271,10 +271,17 @@ func NewTestFactory() *TestFactory {
 		WithClientConfig(clientConfig).
 		WithRESTMapper(testRESTMapper())
 
+	restConfig, err := clientConfig.ClientConfig()
+	if err != nil {
+		panic(fmt.Sprintf("unable to create a fake restclient config: %v", err))
+	}
+
 	return &TestFactory{
 		Factory:           cmdutil.NewFactory(configFlags),
 		FakeDynamicClient: fakedynamic.NewSimpleDynamicClient(legacyscheme.Scheme),
 		tempConfigFile:    tmpFile,
+
+		ClientConfigVal: restConfig,
```

Comment:
@deads2k wiring the restconfig here. The nil pointer panic was being caused because the testFactory's `ToRESTMapper` method returns whatever value is stored here, not the restconfig value from the embedded configFlags